test/functional/lua/diagnostic_spec.lua

```diff
@@ -1654,6 +1667,84 @@ describe('vim.diagnostic', function()
   end)
 
   describe('config()', function()
+    describe('severity_sort', function()
+      before_each(function()
+        exec_lua(function()
+          vim.diagnostic.config({
+            virtual_text = true,
+            underline = true,
+            severity_sort = true,
+          })
+
+          vim.diagnostic.set(_G.diagnostic_ns, _G.diagnostic_bufnr, {
+            _G.make_warning('Some Warning', 0, 0, 0, 0),
+            _G.make_error('Some Error', 0, 0, 0, 0),
+            _G.make_hint('Some Hint', 0, 0, 0, 0),
+          })
+        end)
+      end)
+
+      it('virt_text respects severity_sort reverse=false', function()
+        eq(
+          ' Some Error',
+          exec_lua(function()
+            local extmarks = _G.get_virt_text_extmarks(_G.diagnostic_ns)
+            local virt_text = extmarks[1][4].virt_text
+            return virt_text[5][1]
+          end)
+        )
+      end)
+
+      it('virt_text respects severity_sort reverse=true', function()
+        exec_lua(function()
+          vim.diagnostic.config({
+            severity_sort = { reverse = true },
+          })
+        end)
+
+        eq(
+          ' Some Hint',
+          exec_lua(function()
+            local extmarks = _G.get_virt_text_extmarks(_G.diagnostic_ns)
+            local virt_text = extmarks[1][4].virt_text
+            return virt_text[5][1]
+          end)
+        )
+      end)
```

Comment:
I wrote these tests before realizing that I tried to fixed a working thing, should I remove them or leave it here? 

I think they are not useless